### DISCUSSION TITLE
What is the true weth contract address on mainnet

### DISCUSSION THREAD

**Discussion by FurkanSezal** [zkSync-Community-Hub/zksync-developers#134] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

is there any weth token address on mainnet? I found this one: 0x5AEa5775959fBC2557Cc8789bC1bf90A239D9a91  
looks like the author matterLabs https://explorer.zksync.io/address/0x2ff821c3ca6569140fafce9411049c395f41526a#contract
I found this on syncswap which is also matterLabs product i believe. Should i use this one? Or is there any official weth contract address? 

### What is the specific issue or error you're encountering?

is there any weth token address on mainnet? I found this one: 0x5AEa5775959fBC2557Cc8789bC1bf90A239D9a91  
looks like the author matterLabs https://explorer.zksync.io/address/0x2ff821c3ca6569140fafce9411049c395f41526a#contract

### Can you share the error messages or logs you're receiving, if any?

is there any weth token address on mainnet? I found this one: 0x5AEa5775959fBC2557Cc8789bC1bf90A239D9a91  
looks like the author matterLabs https://explorer.zksync.io/address/0x2ff821c3ca6569140fafce9411049c395f41526a#contract

### Have you made any recent changes to the contract before encountering this issue?

is there any weth token address on mainnet? I found this one: 0x5AEa5775959fBC2557Cc8789bC1bf90A239D9a91  
looks like the author matterLabs https://explorer.zksync.io/address/0x2ff821c3ca6569140fafce9411049c395f41526a#contract

### Are there any external libraries or contracts that your contract interacts with?

is there any weth token address on mainnet? I found this one: 0x5AEa5775959fBC2557Cc8789bC1bf90A239D9a91  
looks like the author matterLabs https://explorer.zksync.io/address/0x2ff821c3ca6569140fafce9411049c395f41526a#contract

### Can you provide the relevant portions of your contract code where the issue is occurring?

is there any weth token address on mainnet? I found this one: 0x5AEa5775959fBC2557Cc8789bC1bf90A239D9a91  
looks like the author matterLabs https://explorer.zksync.io/address/0x2ff821c3ca6569140fafce9411049c395f41526a#contract

### Have you tried to isolate the problem, and if so, what were the results?

is there any weth token address on mainnet? I found this one: 0x5AEa5775959fBC2557Cc8789bC1bf90A239D9a91  
looks like the author matterLabs https://explorer.zksync.io/address/0x2ff821c3ca6569140fafce9411049c395f41526a#contract

### What steps have you already taken to try to resolve the issue?

is there any weth token address on mainnet? I found this one: 0x5AEa5775959fBC2557Cc8789bC1bf90A239D9a91  
looks like the author matterLabs https://explorer.zksync.io/address/0x2ff821c3ca6569140fafce9411049c395f41526a#contract

**Comment by idea404** [comment, marked as answer]
L2 weth address on mainnet is `0x5aea5775959fbc2557cc8789bc1bf90a239d9a91` we'll add a list in future with these addresses of interest to the docs

**Comment by johnwesener** [comment]
Yes, the WETH (Wrapped Ether) token contract address on the Ethereum mainnet is typically 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2. This is the most commonly used WETH contract address across various DeFi protocols and platforms. You can verify this address through Etherscan or other reliable sources. If you found another contract address for WETH, such as the one you mentioned (0x5AEa5775959fBC2557Cc8789bC1bf90A239D9a91), it might be a different implementation or a custom version of WETH provided by specific projects or platforms like zkSync. It's essential to ensure compatibility and integration requirements with the respective platform or project before using a custom WETH contract address.

**Comment by ryangrullon** [comment]
Yes, the WETH (Wrapped Ether) token address on the Ethereum mainnet is 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2. This is the official WETH contract address.

**Comment by gauvinjohnny** [comment]
The WETH token address you found (0x5AEa5775959fBC2557Cc8789bC1bf90A239D9a91) on the zkSync explorer is indeed a wrapped Ethereum (WETH) token deployed by Matter Labs. This address can be used as WETH on the zkSync mainnet. However, it's always a good idea to double-check with official sources or documentation to ensure its validity and compatibility with your specific use case.
